app/code/Magento/ConfigurableProductGraphQl/Model/Resolver/Variant/Variant.php

```diff
@@ -0,0 +1,34 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\ConfigurableProductGraphQl\Model\Resolver\Variant;
+
+use Magento\Framework\GraphQl\Config\Element\Field;
+use Magento\Framework\GraphQl\Query\ResolverInterface;
+use Magento\Framework\GraphQl\Schema\Type\ResolveInfo;
+
+/**
+ * Resolver class for product variant.
+ */
+class Variant implements ResolverInterface
+{
+    /**
+     * @inheritDoc
+     */
+    public function resolve(Field $field, $context, ResolveInfo $info, array $value = null, array $args = null)
+    {
+        if (isset($value['variant']['model']) && $value['variant']['model']) {
+            return
+                array_merge(
```

Comment:
Could you please explain what kind of data do we merge here? An example would help me understand as I don't have a context.

Reply:
Some fields of variant need the product object as model, like \Magento\CatalogGraphQl\Model\Resolver\Product\EntityIdToId. same for image and media_gallery etc...